pkg/apis/components/v1alpha1/types.go

```diff
@@ -38,6 +38,10 @@ type ComponentSpec struct {
 	Metadata     []MetadataItem `json:"metadata"`
 	// +optional
 	InitTimeout string `json:"initTimeout"`
+	// +optional
+	RetryStrategy          string `json:"retryStrategy"`
```

Comment:
ComponentSpec is not pubsub specific. These values should come from Metadata instead.

Reply:
Hi @artursouza Thanks for the comments. Will look and start updating the PR. Specifically for this one on the retry settings, can you please look at the issue's thread with @yaron2. He suggested doing it this way instead of using the metadata.

Reply:
See https://github.com/dapr/dapr/issues/3000#issuecomment-813749122

Reply:
@juazasan Thanks for the reminder on this. I am taking another look at this PR with this in mind.

Reply:
Sure. I will be reviewing the results of the e2e tests in the meanwhile. I found out some bugs in the retry module that I am  fixing, and also trying to figure it out how to address the issues with the grpc implementation as they seem to introduce a breaking change. Thanks.

Reply:
@yaron2 Can we revisit this decision?

Reply:
If we use metadata and there are bugs in the retry stuff, I recommend using the [retry package that exists in components-contrib](https://github.com/dapr/components-contrib/tree/master/internal/retry) as it has become a pattern that many of the components have adopted for configurable retry. And the idea is that it will be the standard across Dapr. And if there are issues with it, I'd rather address them there so it benefits everything that uses it. I'd be happy to move it to dapr/kit now if that would help. Here is what the code would look like.

```golang
// Default retry configuration is used if no
// backOff properties are set.
if err := retry.DecodeConfigWithPrefix(
	&p.backOffConfig,
	metadata.Properties,
	"backOff"); err != nil {
	return err
}
```